MessageViewController/MessageAutocompleteController.swift

```diff
@@ -80,12 +101,34 @@ public final class MessageAutocompleteController: MessageTextViewListener {
         )
 
         guard let range = Range(insertionRange, in: text) else { return }
+        
+        // Apply the autocomplete attributes
+        var attrs = autocompleteTextAttributes[selection.prefix] ?? defaultTextAttributes
+        attrs[NSAttributedAutocompleteKey] = true
+        let newString = (keepPrefix ? selection.prefix : "") + autocomplete
+        let newAttributedString = NSAttributedString(string: newString, attributes: attrs)
+        
+        // Create an NSRange to use with attributedText replacement
+        let nsrange = NSRange(range, in: textView.text)
+        
+        // Modify the NSRange to include the prefix length
+        let rangeModifier = keepPrefix ? selection.prefix.count : 0
+        let highlightedRange = NSRange(location: nsrange.location - rangeModifier, length: nsrange.length + rangeModifier)
+        
+        // Replace the attributedText with a modified version including the autocompete
+        if appendSpaceOnCompletion {
+            textView.attributedText = textView.attributedText.replacingCharacters(in: highlightedRange, with: newAttributedString) + NSAttributedString(string: " ", attributes: typingTextAttributes)
+        } else {
+            textView.attributedText = textView.attributedText.replacingCharacters(in: highlightedRange, with: newAttributedString)
+        }
```

Comment:
style nit: can we put params on newlines to keep the columns <120?

It looks like the line

```swift
textView.attributedText.replacingCharacters(in: highlightedRange, with: newAttributedString)
```

Is included twice. Could you put that in a var before entering the if statement? 

Or simply:

```swift
textView.attributedText = textView.attributedText.replacingCharacters(in: highlightedRange, with: newAttributedString)
if appendSpaceOnCompletion {
  textView.attributedText += NSAttributedString(string: " ", attributes: typingTextAttributes)
}
```

Reply:
Reminder: SwiftLint

Reply:
👍 Private API function?